hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java

```diff
@@ -403,6 +360,34 @@ protected ByteString readChunk(final ChunkInfo chunkInfo,
     return xceiverClient.getPipeline().getNodes();
   }
 
+  private CheckedFunction<ContainerProtos.ContainerCommandResponseProto, IOException>
+      validator = (response) -> {
+    ReadChunkResponseProto readChunkResponse;
+    final ChunkInfo chunkInfo = chunks.get(chunkIndex);
+    ByteString byteString;
+    try {
+      ContainerProtocolCalls.validateContainerResponse(response);
```

Comment:
Instead of calling the another validate function inside a validator, we can pass a list of validators to XceiverClientSpi#sendCommand().

Reply:
Addressed in the latest patch.